server/infoextractor.ts

```diff
@@ -264,25 +265,35 @@ export default {
 							.filter(p => !video[p]),
 					}))
 					.filter(request => request.missingInfo.length > 0);
-
 				if (requests.length === 0 && adapter.isCacheSafe) {
 					return cachedVideos;
 				}
 
 				const fetchedVideos = await adapter.fetchManyVideoInfo(requests);
-				const finalResults = cachedVideos.map(video => {
-					const fetchedVideo = fetchedVideos.find(v => v.id === video.id);
-					if (fetchedVideo) {
-						return mergeVideo(video, fetchedVideo);
-					} else {
-						return video;
-					}
-				});
+				const finalResults = cachedVideos
+					.map(video => {
+						const fetchedVideo = fetchedVideos.find(v => v.id === video.id);
+						const videoKeys = Object.keys(video);
+						if (fetchedVideo) {
+							return mergeVideo(video, fetchedVideo);
+						} else if (videoKeys.length > 2) {
+							return video;
```

Comment:
What is this branch here for?

Reply:
You're right it's still broken. I don't think this approach is correct, I'll revise it.

The branch is to only return the video if the video is found in the cache, since it should contain more than the VideoId fields then.

Reply:
This is incorrect behavior then. We need to be able to return videos that were not already in the cache.